quartz/components/scripts/popover.inline.ts

```diff
@@ -82,8 +82,6 @@ async function mouseEnterHandler(
       const contents = await response.text()
       const html = p.parseFromString(contents, "text/html")
       normalizeRelativeURLs(html, targetUrl)
-      // strip all IDs from elements to prevent duplicates
-      html.querySelectorAll("[id]").forEach((el) => el.removeAttribute("id"))
```

Comment:
we need this to prevent regressing on anchors to the same page not scrolling to the real header (if the id is the same, it will scroll to the header in the popover rather than the real one)

Reply:
Scroll in popovers is currently broken because it only works with `id`s. By stripping them out, popovers will never scroll to their linked anchor.

Would any of the following work for you?:

- disabling popovers to links on the current page.
- not disabling popovers to the current page, but stripping `id` if current page? (This will break popover scroll to anchor in those cases).

Reply:
we can maybe map the ids inside the popover to be like popover-${id} and change the scroll in popover behavior to scroll to the modified id

Reply:
How would that interact with cross-page anchors? An element can only have a single `id`.

Reply:
that doesnt change anything? im saying map the ids of all elements in the popover so they are namespaced by the popover (otherwise we get id collisions)

Reply:
Never mind, I have an idea. Gimme a bit.

Reply:
I have found a proper fix. I have left a few examples on the `configuration` page for quick testing. Please let me know if it also works on your end. I'll remove them before merging.

Reply:
can you update the comment? we aren't stripping them anymore, we are just mapping them

additionally, this makes sense to do even on cross-page links (e.g. you are on page A linking to page B#C but page A also has a #C heading)